documentation/polymer-templates/tutorial-template-intro.asciidoc

```diff
@@ -1,59 +1,44 @@
 ---
-title: Using Polymer Templates from Server-side Java Code
+title: Using Templates from Server-side Java Code
 order: 1
 layout: page
 ---
 
 ifdef::env-github[:outfilesuffix: .asciidoc]
 
-= Using Polymer Templates from Server-side Java Code
+= Using Templates from Server-side Java Code
 
-Polymer (HTML) template support in Vaadin allows you to create pre-formatted HTML components (declarative layouts) that can be used from Java in the same way as any other component. Polymer templates give you full control over the appearance (look) of the DOM. You can use any Vaadin component in your design, as well as JavaScript logic, and even CSS. Declarative layouts have many benefits, but there are also disadvantages.
+In Vaadin, you have a few different options on how to build your views.
 
-== When to Use Polymer Templates
+. Build your views purely with Java.
+. Define your layouts declaratively and implement the UI logic in Java.
+. Build your views purely client-side with TypeScript and LitElement. Available from Vaadin 15 forward.
 
-Polymer templates work best:
+These ways to build views can be mixed and matched within an application. All Vaadin components are available through Java API as well as declaratively in HTML. Choosing the correct way depends on your use case and preferences. This chapter goes through the second option, how to build your UIs declaratively while implementing the logic in Java.
```

Comment:
Maybe it would be good to mention that you can also add your own application-specific components (designed in either of the three ways) as building blocks in the templates.

Reply:
Not sure what kind of sentence would be added. How do you add a component done with Java into a template? Can you suggest a change to the text?

Reply:
I assume you would use `<slot>` in the template and then `Component` or `Element` API on the server side to insert Java components there. Maybe this detail should not be mentioned here, but may be important to somehow ensure the users that client-side components are not "leaves" (and also that you can have a recursive component structure even though it is static HTML).

Reply:
Maybe a sentence like "Regardless of which approach you build your view in, it can be composed of both Java components and template-based components." would fit the bill.

Reply:
I added a comment about this under the header "Benefits of Using Templates" as it was a thing that applied to templates specifically.

> Java-only components like compositions built with Java can be added to a HTML template by giving a layout an id in the template and adding the Java component in Java code.